packages/console/src/pages/UserDetails/UserSettings/components/PersonalAccessTokens/CreateTokenModal.tsx

```diff
@@ -0,0 +1,148 @@
+import { type PersonalAccessToken } from '@logto/schemas';
+import { addDays, format } from 'date-fns';
+import { useCallback, useEffect, useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { toast } from 'react-hot-toast';
+import { useTranslation } from 'react-i18next';
+import ReactModal from 'react-modal';
+
+import Button from '@/ds-components/Button';
+import DangerousRaw from '@/ds-components/DangerousRaw';
+import FormField from '@/ds-components/FormField';
+import ModalLayout from '@/ds-components/ModalLayout';
+import Select from '@/ds-components/Select';
+import TextInput from '@/ds-components/TextInput';
+import useApi from '@/hooks/use-api';
+import modalStyles from '@/scss/modal.module.scss';
+import { trySubmitSafe } from '@/utils/form';
+
+type FormData = { name: string; expiration: string };
+
+type Props = {
+  readonly userId: string;
+  readonly isOpen: boolean;
+  readonly onClose: (createdToken?: PersonalAccessToken) => void;
+};
+
+const days = Object.freeze([7, 30, 180, 365]);
+const neverExpires = '-1';
+
+function CreateTokenModal({ userId, isOpen, onClose }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const {
+    register,
+    control,
+    watch,
+    formState: { errors, isSubmitting },
+    handleSubmit,
+    reset,
+  } = useForm<FormData>({ defaultValues: { name: '', expiration: String(days[0]) } });
+  const onCloseHandler = useCallback(
+    (created?: PersonalAccessToken) => {
+      reset();
+      onClose(created);
+    },
+    [onClose, reset]
+  );
+  const api = useApi();
+  const expirationDays = watch('expiration');
+  const [expirationDate, setExpirationDate] = useState<Date>();
+
+  useEffect(() => {
+    const setDate = () => {
+      if (expirationDays === neverExpires) {
+        setExpirationDate(undefined);
+      } else {
+        setExpirationDate(addDays(new Date(), Number(expirationDays)));
+      }
+    };
```

Comment:
Update the this state in the `onChange` callback from the `Select` component?

Reply:
There is an interval to clear, so `useEffect` would be easier.